pkg/util/merge/merge_service_spec.go

```diff
@@ -5,13 +5,70 @@ import (
 )
 
 // ServiceSpec merges two ServiceSpecs together.
-// The implementation does not override:
-// - labels/selectors
-// - cluster IPs
+// The implementation merges Selector instead of overriding it.
 func ServiceSpec(defaultSpec, overrideSpec corev1.ServiceSpec) corev1.ServiceSpec {
-	mergedSpec := overrideSpec
+	mergedSpec := defaultSpec
 	mergedSpec.Selector = StringToStringMap(defaultSpec.Selector, overrideSpec.Selector)
-	mergedSpec.ClusterIPs = defaultSpec.ClusterIPs
-	mergedSpec.ClusterIP = defaultSpec.ClusterIP
+
+	if len(overrideSpec.Ports) != 0 {
```

Comment:
I'm not sure this is the right approach, because this would prevent us from actually setting zero-value in override. 
For IP or Type field this is not a problem, but field like PublishNotReadyAddresses would never set it to false.

Reply:
That's a fair point. We could leave this as it was before. The only problem is that you won't be able to do cascading overrides in MongoDBMulti, such as:

1. use `PublishNotReadyAddresses` as `true` by default
2. override type and set it to `loadbalancer` in the `spec`
3. override individual Load Balancer IPs in specific clusterSpecs. 

You will have to always specify all 3 properties (`PublishNotReadyAddresses`, Load Balancer type and Load Balancer IPs) in the clusterSpec. I think that's fine, although - a bit counter intuitive. But if anyone complains about it, we could enhance this in the future.

Reply:
I'm slightly for less convenient approach, but without surprises, so as it was before.